core/domain/user_domain_test.py

```diff
@@ -0,0 +1,353 @@
+            feconf.DEFAULT_FEEDBACK_NOTIFICATIONS_MUTED_PREFERENCE)
+        default_suggestion_notifs = (
+            feconf.DEFAULT_SUGGESTION_NOTIFICATIONS_MUTED_PREFERENCE)
+
+        test_dict = self.test_prefs_email.to_dict()
+        default_dict = self.default_prefs_email.to_dict()
+        self.assertEqual(
+            test_dict,
+            {
+                'mute_feedback_notifications': False,
+                'mute_suggestion_notifications': True
+            }
+        )
+        self.assertEqual(
+            default_dict,
+            {
+                'mute_feedback_notifications':
+                default_feedback_notifs,
+                'mute_suggestion_notifications':
+                default_suggestion_notifs
+            }
+        )
+
+
+class ExpUserLastPlaythroughTests(test_utils.GenericTestBase):
+    """Testing domain object for an exploration last playthrough model."""
+
+    def setUp(self):
+        super(ExpUserLastPlaythroughTests, self).setUp()
+        self.exp_last_playthrough = (user_domain.ExpUserLastPlaythrough(
+            'user_id0', 'exp_id0', 0, 'last_updated', 'state0'))
+
+    def test_initialization(self):
+        """Testing init method."""
+        self.assertEqual(
+            self.exp_last_playthrough.id, 'user_id0.exp_id0')
+        self.assertEqual(
+            self.exp_last_playthrough.user_id, 'user_id0')
+        self.assertEqual(
+            self.exp_last_playthrough.exploration_id, 'exp_id0')
+        self.assertEqual(
+            self.exp_last_playthrough.last_played_exp_version, 0)
+        self.assertEqual(
+            self.exp_last_playthrough.last_updated, 'last_updated')
+        self.assertEqual(
+            self.exp_last_playthrough.last_played_state_name, 'state0')
+
+    def test_update_last_played_information(self):
+        """Testing update_last_played_information."""
+        self.exp_last_playthrough.update_last_played_information(1, 'state1')
+        self.assertEqual(
+            self.exp_last_playthrough.last_played_exp_version, 1)
+        self.assertEqual(
+            self.exp_last_playthrough.last_played_state_name, 'state1')
+
+
+class IncompleteActivitiesTests(test_utils.GenericTestBase):
+    """Testing domain object for incomplete activities model."""
+
+    def setUp(self):
+        super(IncompleteActivitiesTests, self).setUp()
+        self.incomplete_activities = (user_domain.IncompleteActivities(
+            'user_id0', ['exp_id0'], ['collect_id0']))
+
+    def test_initialization(self):
+        """Testing init method."""
+        expected_exp_list = ['exp_id0']
+        expected_coll_list = ['collect_id0']
+
+        self.assertEqual(self.incomplete_activities.id, 'user_id0')
+        self.assertListEqual(
+            self.incomplete_activities.exploration_ids, expected_exp_list)
+        self.assertListEqual(
+            self.incomplete_activities.collection_ids, expected_coll_list)
+
+    def test_add_exploration_id(self):
```

Comment:
Generally when doing an action, the pattern is:
- do a baseline test 
- do the action
- do an endline test that's in the same format as the baseline

This test is missing a baseline test, so let's add it. Similarly below.

Reply:
I redid the tests to fit the pattern. The tests for the actions now follow the pattern of baseline, action, endline testing. 

Thank you for teaching me this.